addons/sale/sale.py

```diff
@@ -616,7 +615,7 @@ def _action_procurement_create(self):
 
     @api.model
     def _get_analytic_invoice_policy(self):
-        return ['cost']
+        return ['delivery']
```

Comment:
yes you are right. we should remove overrided create method of sale.order.line from sale module.

and In sale_timesheet module, override create/write method of sale.order.line and call create_analytic_account base on track service and product type.

and In sale_expense module, override create/write method of sale.order.line and call create_analytic_account base on invoice policy.